Add usage example for lines_from function (rectangle)

Samwong077 opened the pull request:
# Overview
This example demonstrates the use of the lines_from function in SplashKit to extract and interact with the edges of a rectangle. It visualizes how each edge can be individually styled and responds dynamically to mouse hover events. This helps developers understand not just what lines_from returns, but also how to use it in interactive graphics applications.

SplashKit Function: lines_from
# Files Included
- [x] Relevant title for the example (.txt)
- [x] C++ code
- [x] C# code (Top-Level statements)
- [x] C# code (Object-Oriented Programming)
- [x] Python code
- [x] gif
- [x] .txt

# Usage Example Checks
- [x] Simple, clear demonstration of the function
- [x] Code uses Splashkit functions
- [x] Tested in Chrome and Firefox

nielsent478 commented:
Hey Sam, I have tested the code and the code runs as what was displayed in the 'line-from-1-exmaple.gif' fine. However I have noticed that for each program file, more code comments should be used to help with the readability of the code. I have also jotted down some improvements that could be used and good features.

 C# OOP and Top-level, C++ and Python programs should all use necessary code comments (Listed in the comments below)
 
Improvements:

- [] File namings should be 'line-from-with-rectangle-1-example' instead of 'line-from-1-exmaple' instead of 'cicle_intersects_triangle-1-example'.
- []Lines could be bigger so it can be easily hovered

Good notes:
- [x] Good txt file documentation
- [x] Program output as been tested:

![image](https://github.com/user-attachments/assets/ac431bb3-b9a2-4971-96b0-05a5b15e5210)

Please note, the changes requested below for the comments are only in the OOP version, but please apply the relevant changes to every version of the program.

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+    private List<Line> rectangleEdges;
+    private List<Color> normalColors;
+    private List<Color> hoverColors;
+
+    public void Run()
```
Explain here how this is called

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+
+public class Program
+{
+    public static void Main()
+    {
+        new RectangleLinesApp().Run();
+    }
```
Comment here for the call to the run() in the rectanglelinesapp class

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+
+    public void Run()
+    {
+        window = new Window("Lines From Rectangle Example", 800, 600);
+        myRect = SplashKit.RectangleFrom(250, 200, 300, 200);
```
Comment here about the new window and the rectanglefrom

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+    {
+        window = new Window("Lines From Rectangle Example", 800, 600);
+        myRect = SplashKit.RectangleFrom(250, 200, 300, 200);
+        rectangleEdges = SplashKit.LinesFrom(myRect);
```
Here for the linesfrom for the rectangle edges

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+        while (!window.CloseRequested)
+        {
+            SplashKit.ProcessEvents();
+            SplashKit.ClearScreen(Color.DarkSlateGray);
+            SplashKit.FillRectangle(Color.Black, myRect.X, myRect.Y, myRect.Width, myRect.Height);
```
Explain what these two methods perform

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+            SplashKit.ClearScreen(Color.DarkSlateGray);
+            SplashKit.FillRectangle(Color.Black, myRect.X, myRect.Y, myRect.Width, myRect.Height);
+
+            Point2D mouse = SplashKit.MousePosition();
```
Comments how this handles from the mouse positon

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+
+            Point2D mouse = SplashKit.MousePosition();
+
+            for (int i = 0; i < rectangleEdges.Count; i++)
```
Add comments here

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+
+            for (int i = 0; i < rectangleEdges.Count; i++)
+            {
+                Line edge = rectangleEdges[i];
```
comments here of the stored index

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+            {
+                Line edge = rectangleEdges[i];
+
+                if (SplashKit.PointOnLine(mouse, edge))
```
Explain what this does

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+                }
+                else
+                {
+                    SplashKit.DrawLine(normalColors[i], edge);
```
Same over here

nielsent478 commented on public/usage-examples/graphics/line-from-1-exmaple-oop.cs:
```diff
@@ -0,0 +1,65 @@
+        SplashKit.DrawLine(
+            color,
+            edge.StartPoint.X, edge.StartPoint.Y,
+            edge.EndPoint.X, edge.EndPoint.Y
```
Explain this and each argument as a comment

nielsent478 reviewed (changes requested):
Overall, everything works as expected just needs the comment suggestions above from the file addressed for each code file, and then I can approve of this request.

omckeon reviewed (changes requested):
Please check the other text files for the format to use - should be only 1 line.

And there are a lot of issues with the code files (that have not been mentioned in the review). @Samwong077 - Please check the Style guide and fix up all of your PRs, and all your reviews by this coming Wednesday 14th May.

@nielsent478 - Please check the Style guide as well and then redo this review once @Samwong077 has fixed the code.

Samwong077 commented:
> Overall, everything works as expected just needs the comment suggestions above from the file addressed for each code file, and then I can approve of this request.

I've updated the code, and fixed the file location. Also I forgot to delete the original wrong path code, so I am going to delete it now.

nielsent478 commented on public/usage-examples/geometry/line_from-1-example-oop.cs:
```diff
@@ -0,0 +1,108 @@
+using SplashKitSDK;
+
+// Entry point class for the program
+public class Program
+{
+    public static void Main()
```
Follow the usage-example style guide here

nielsent478 commented on public/usage-examples/geometry/line_from-1-example-oop.cs:
```diff
@@ -0,0 +1,108 @@
+    public static void Main()
+    {
+        // Create a new instance of the app and run it
+        RectangleLinesApp app = new RectangleLinesApp();
```
Move the class into the main loop

nielsent478 commented on public/usage-examples/geometry/line_from-1-example-top-level.cs:
```diff
@@ -0,0 +1,78 @@
+using static SplashKitSDK.SplashKit;
```
Only needs this

nielsent478 commented:
I have reviewed this pull request for lines_from_function (rectangle) and provided code comments for what I think is necessary to be changed.

As a summary, Please follow the usage-example Style Guide to ensure that all code files align to it's documentation. Especially with the headers and only using open_window. As well as the OOP code especially following the usage-example structure.

Comments can be simplified so it is more understandable to new SIT102 students.

Case conventions are fine.


After these changes, I will be able to approve of this pull request.

nielsent478 reviewed (changes requested):
- Forgot to leave the requested changes

Samwong077 commented:
> * Forgot to leave the requested changes

I have updated the new code.

nielsent478 commented:
Hey Sam, I've looked through each code file and it seems mostly fine. However I've noticed a few things to address that still has not yet been done:

- Ensure proper naming conventions according to the style guide for each code file
- Only use one class (other code not in a separate class) for the OOP program code and the python one doesn't need that main() call.

Everything else looks good and the programs ran fine.

After these changes I can approve this

nielsent478 commented on public/usage-examples/geometry/lines_from-1-example-oop.cs:
```diff
@@ -0,0 +1,63 @@
+        // Open the window using SplashKit.OpenWindow to match usage guide
+        SplashKit.OpenWindow("Interactive Rectangle Edges", 800, 600);
+
+        _myRectangle = SplashKit.RectangleFrom(250, 200, 300, 200);
+        _rectangleEdges = SplashKit.LinesFrom(_myRectangle);
+
+        _normalColors = new List<Color> { Color.Red, Color.LimeGreen, Color.Blue, Color.Orange };
+        _hoverColors = new List<Color> { Color.White, Color.White, Color.White, Color.White };
```
Use proper naming conventions from the style guide

nielsent478 commented on public/usage-examples/geometry/lines_from-1-example.py:
```diff
@@ -0,0 +1,53 @@
+              edge.start_point.x, edge.start_point.y,
+              edge.end_point.x, edge.end_point.y)
+
+def main():
```
unnecessary main

nielsent478 commented on public/usage-examples/geometry/lines_from-1-example.py:
```diff
@@ -0,0 +1,53 @@
+
+    close_window(window)
+
+main()
```
unnecessary main call

nielsent478 reviewed (changes requested):
Please note that these changes should be applied to each file when necessary and make sure that the file names are 'example' not 'exmaple' especially in the old files
